open_rarity/models/collection.py

```diff
@@ -231,5 +261,37 @@ def extract_collection_attributes(
 
         return collection_traits
 
+    def _derive_normalized_attributes_frequency_counts(
+        self,
+    ) -> dict[AttributeName, dict[AttributeValue, int]]:
+        """Derives and constructs attributes_frequency_counts based on
+        string attributes on tokens. Numeric or date attributes currently not
+        supported.
+
+        Returns
+        -------
+        dict[ AttributeName, dict[AttributeValue, int] ]
+            dictionary of attributes to the number of tokens in this collection
+            that has a specific value for every possible value for the given
+            attribute, by default None.
+        """
+        attrs_freq_counts: dict[
+            AttributeName, dict[AttributeValue, int]
+        ] = defaultdict(dict)
+
+        for token in self._tokens:
+            for (
+                attr_name,
+                str_attr,
+            ) in token.metadata.string_attributes.items():
+                normalized_name = normalize_attribute_string(attr_name)
+                normalized_value = normalize_attribute_string(str_attr.value)
```

Comment:
Can we assume `name` and `value` are already normalized?

Should we use the exact name and value stored in the attribute to guarantee consistency?

Reply:
Check new format